php/class-terminus-command.php

```diff
@@ -1,66 +1,93 @@
 <?php
+
+use \Terminus\Auth;
 use \Terminus\Endpoint;
 use \Terminus\Request;
 use \Terminus\Session;
-use \Terminus\Auth;
 use \Terminus\Loggers\Regular as Logger;
 
 /**
  * Base class for Terminus commands
  *
- * @package terminus
+ * @package Terminus
  */
 abstract class Terminus_Command {
-
   public $cache;
   public $session;
   public $sites;
 
-  protected $_func;
-  protected $_siteInfo;
-  protected $_bindings;
-  protected static $_blacklist = array('password');
+  protected static $blacklist = array('password');
+  protected $func;
 
+  /**
+   * Instantiates object, sets cache and session
+   *
+   * @return [Terminus_Command] $this
+   */
   public function __construct() {
-    # Load commonly used data from cache.
-    $this->cache = Terminus::get_cache();
+    //Load commonly used data from cache
+    $this->cache   = Terminus::get_cache();
     $this->session = Session::instance();
   }
 
   /**
-   * Make a request to the Dashbord's internal API.
+   * Downloads the given URL to the given target
    *
-   * @param $realm
-   *    Permissions realm for data request: currently "user" or "site" but in the
-   *    future this could also be "organization" or another high-level business
-   *    object (e.g. "product" for managing your app). Can also be "public" to
-   *    simply pull read-only data that is not privileged.
-   *
-   * @param $uuid
-   *    The UUID of the item in the realm you want to access.
-   *
-   * @param $method
-   *    HTTP method (verb) to use.
+   * @param [string] $url    Location of file to download
+   * @param [string] $target Location to download file to
+   * @return [void]
+   */
+  public static function download($url, $target) {
+    try {
+      $response = Request::download($url, $target);
+      return $target;
+    } catch(Exception $e) {
+      Terminus::error($e->getMessage());
+    }
+  }
+
+  /**
+   * Make a request to the Pantheon API
    *
-   * @param $data
-   *    A native PHP data structure (int, string, arary or simple object) to be
-   *    sent along with the request. Will be encoded as JSON for you.
+   * @param [string] $realm   Permissions realm for data request (e.g. user,
+   *   site organization, etc. Can also be "public" to simply pull read-only
+   *   data that is not privileged.
+   * @param [string] $uuid    The UUID of the item in the realm to access
+   * @param [string] $path    API path (URL)
+   * @param [string] $method  HTTP method to use
+   * @param [mixed]  $options A native PHP data structure (e.g. int, string,
+   *   array, or stdClass) to be sent along with the request
+   * @return [array] $data
    */
-  public static function request($realm, $uuid, $path = FALSE, $method = 'GET', $options = NULL) {
-    if (!in_array($realm, array('login','user','public'))) {
+  public static function request(
+    $realm,
+    $uuid,
+    $path = false,
+    $method = 'GET',
+    $options = null
+  ) {
```

Comment:
What is the rule for splitting arguments over multiple lines? It seems like 2 arg methods aren't being split (+1), but just wanted to know for my info.